system/jlib/jmutex.hpp

```diff
@@ -288,7 +288,15 @@ class CriticalSection
     inline ~CriticalSection()
     {
 #ifdef _ASSERT_LOCK_SUPPORT
-        assertex(owner==0 && depth==0);
```

Comment:
On modern compilers try/catch should add no overhead to the code.
I could convert it to always use a fprintf instead, but potentially throwing an exception in a destructor isn't a good idea.